src/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { Principal } from "@dfinity/principal";
+import { IndefiniteEvent, DetailValue } from "./declarations/cap/root";
+
+type DetailType =
+  | Principal
+  | bigint
+  | boolean
+  | string
+  | number
+  | Array<number>
+  | Array<DetailType>;
+type PrettyDetail = { [key: string]: DetailType };
+type DetailVector = { Vec: Array<DetailValue> };
+type GeneralValue = { [key: string]: any };
+
+const decodeDetailValue = (value: DetailValue): DetailType => {
+  const type = Object.keys(value)?.[0];
+  switch (type) {
+    case "Vec":
+      return (value as DetailVector)?.["Vec"]?.map(decodeDetailValue);
+    case "True":
+      return true;
+    case "False":
+      return false;
+    default:
+      return (value as GeneralValue)?.[type];
+  }
+};
+
+export const preattifyCapTransactions = (transaction: IndefiniteEvent) => {
```

Comment:
```suggestion
export const prettifyCapTransactions = (transaction: IndefiniteEvent) => {
```

Reply:
Also, would suggest exporting from entry point, so that's easily imported from `@psychedelic/cap-js` scope instead of `@psychedelic/cap-js/src/utils`.